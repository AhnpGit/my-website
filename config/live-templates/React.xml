<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
