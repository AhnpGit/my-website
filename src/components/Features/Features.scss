@import '../../assets/sass/abstracts/variables';
@import '../../assets/sass/abstracts/mixins';

.features__intro {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  h2 {
    text-align: center;
    word-spacing: 2px;
    letter-spacing: 1px;

    &:after {
      display: block;
      height: 2px;
      background-color: map_get($colors, tahitiGold);
      content: " ";
      width: 100px;
      margin: 0 auto;
      margin-top: map_get($base, 'unit');
    }
  }

  p {
    max-width: 700px;
    margin-bottom: 0;
    line-height: 145%;
    text-align: center;
  }

  margin-bottom: map_get($base, unit);
}

@lost clearing both;
@lost rounder 99.9;
.features__specs {
  lost-center: 100% 0 flex;
  .info {
    lost-row: 1/4 0 map_get($base, 'unit');
  }

  @include mq(nth($medium-up, 1)) {
    .info {
      lost-column: 1/4 0 map_get($base, 'unit');
    }
  }

  h3 {
    text-transform: uppercase;
  }
}
