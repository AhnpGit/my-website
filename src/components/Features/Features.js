// @flow

import React from 'react'
import CustomPropTypes from '../../utils/CustomPropTypes'
import type { FeaturesType } from '../../utils/types'

import InfoItem from '../../elements/InfoItem/InfoItem'
import Row from '../../elements/Row/Row'
import Wrapper from '../../elements/Wrapper/Wrapper'

import './Features.scss'

/**
 * Feature component displays a title, an introduction and 4 feature items
 */
const Features = ({ featureItems }: FeaturesType) => (
  <Row className="features">
    <Wrapper name="features">
      <div className="features__intro">
        <h2>Get food fast &mdash; not fast food</h2>
        <p>
          Hello, we're Omnifood, your new premium food delivery service. We know you're always busy. No time for cooking.
          So let us take care of that, we're really good at it, we promise!
        </p>
      </div>

      <div className="features__specs">
        {featureItems.map((featureItem) => <InfoItem key={featureItem.id} title={featureItem.title} text={featureItem.text} />)}
      </div>
    </Wrapper>
  </Row>
)

Features.propTypes = {
  /**
   * Array which contains 4 feature items
   */
  featureItems: CustomPropTypes.array(4).isRequired
}

export default Features
