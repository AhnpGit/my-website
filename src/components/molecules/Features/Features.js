// @flow

import React from 'react'
import CustomPropTypes from '../../../utils/CustomPropTypes'

import GridItem from '../../atoms/GridItem/GridItem'
import type { FeaturesType } from '../../../utils/types'

/**
 * Feature component displays a title, an introduction and 4 feature items
 */
const Features = ({ featureItems } : FeaturesType) => (
  <section className="features">
    <div className="features__intro">
      <h2>Get food fast &mdash; not fast food</h2>
      <p>
        Hello, we're Omnifood, your new premium food delivery service. We know you're always busy. No time for cooking.
        So let us take care of that, we're really good at it, we promise!
      </p>
    </div>

    <div className="features__specs">
      {featureItems.map((featureItem) => <GridItem key={featureItem.id} gridItem={featureItem} className="specs__span--1--of--4" />)}
    </div>
  </section>
)

Features.propTypes = {
  /**
   * Array which contains 4 feature items
   */
  featureItems: CustomPropTypes.array(4).isRequired
}

export default Features
