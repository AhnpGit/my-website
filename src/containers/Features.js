// @flow

import { connect } from 'react-redux'
import Features from '../components/Features/Features'
import type { FeaturesType } from '../utils/types'

type FeaturesState = {
  features: FeaturesType
}

export const mapStateToProps = (state : FeaturesState) => ({
  featureItems: state.features.featureItems
})

export default connect(
  mapStateToProps
)(Features)
