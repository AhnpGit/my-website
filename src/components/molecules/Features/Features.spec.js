import React from 'react'
import Features from './Features'
import { shallow } from 'enzyme'

describe('Hero component', () => {
  it('Features correctly without issues', () => {
    const featureItems = [
      {
        id: 1,
        title: "title 1",
        text: "text 1"
      },
      {
        id: 2,
        title: "title 2",
        text: "text 2"
      },
      {
        id: 3,
        title: "title 3",
        text: "text 3",
        image: {
          url: 'Example image url',
          alt: "Image 1"
        }
      },
      {
        id: 4,
        title: "title 4",
        text: "text 4",
        image: {
          url: 'Example image url',
          alt: "Image 2"
        }
      },
    ]

    const result = shallow(
      <Features featureItems={featureItems} />
    )

    expect(result).toMatchSnapshot()
  })
})
