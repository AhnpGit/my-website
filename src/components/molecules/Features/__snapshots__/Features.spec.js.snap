// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component Features correctly without issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="features"
>
    <div
        className="features__intro"
    >
        <h2>
            Get food fast — not fast food
        </h2>
        <p>
            Hello, we're Omnifood, your new premium food delivery service. We know you're always busy. No time for cooking. So let us take care of that, we're really good at it, we promise!
        </p>
    </div>
    <div
        className="features__specs"
    >
        <GridItem
            className="specs__span--1--of--4"
            gridItem={
                Object {
                    "id": 1,
                    "text": "text 1",
                    "title": "title 1",
                  }
            }
        />
        <GridItem
            className="specs__span--1--of--4"
            gridItem={
                Object {
                    "id": 2,
                    "text": "text 2",
                    "title": "title 2",
                  }
            }
        />
        <GridItem
            className="specs__span--1--of--4"
            gridItem={
                Object {
                    "id": 3,
                    "image": Object {
                      "alt": "Image 1",
                      "url": "Example image url",
                    },
                    "text": "text 3",
                    "title": "title 3",
                  }
            }
        />
        <GridItem
            className="specs__span--1--of--4"
            gridItem={
                Object {
                    "id": 4,
                    "image": Object {
                      "alt": "Image 2",
                      "url": "Example image url",
                    },
                    "text": "text 4",
                    "title": "title 4",
                  }
            }
        />
    </div>
</section>,
  "nodes": Array [
    <section
      className="features"
>
      <div
            className="features__intro"
      >
            <h2>
                  Get food fast — not fast food
            </h2>
            <p>
                  Hello, we're Omnifood, your new premium food delivery service. We know you're always busy. No time for cooking. So let us take care of that, we're really good at it, we promise!
            </p>
      </div>
      <div
            className="features__specs"
      >
            <GridItem
                  className="specs__span--1--of--4"
                  gridItem={
                        Object {
                              "id": 1,
                              "text": "text 1",
                              "title": "title 1",
                            }
                  }
            />
            <GridItem
                  className="specs__span--1--of--4"
                  gridItem={
                        Object {
                              "id": 2,
                              "text": "text 2",
                              "title": "title 2",
                            }
                  }
            />
            <GridItem
                  className="specs__span--1--of--4"
                  gridItem={
                        Object {
                              "id": 3,
                              "image": Object {
                                "alt": "Image 1",
                                "url": "Example image url",
                              },
                              "text": "text 3",
                              "title": "title 3",
                            }
                  }
            />
            <GridItem
                  className="specs__span--1--of--4"
                  gridItem={
                        Object {
                              "id": 4,
                              "image": Object {
                                "alt": "Image 2",
                                "url": "Example image url",
                              },
                              "text": "text 4",
                              "title": "title 4",
                            }
                  }
            />
      </div>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Features
        featureItems={
                Array [
                        Object {
                          "id": 1,
                          "text": "text 1",
                          "title": "title 1",
                        },
                        Object {
                          "id": 2,
                          "text": "text 2",
                          "title": "title 2",
                        },
                        Object {
                          "id": 3,
                          "image": Object {
                            "alt": "Image 1",
                            "url": "Example image url",
                          },
                          "text": "text 3",
                          "title": "title 3",
                        },
                        Object {
                          "id": 4,
                          "image": Object {
                            "alt": "Image 2",
                            "url": "Example image url",
                          },
                          "text": "text 4",
                          "title": "title 4",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "featureItems": Array [
            Object {
              "id": 1,
              "text": "text 1",
              "title": "title 1",
            },
            Object {
              "id": 2,
              "text": "text 2",
              "title": "title 2",
            },
            Object {
              "id": 3,
              "image": Object {
                "alt": "Image 1",
                "url": "Example image url",
              },
              "text": "text 3",
              "title": "title 3",
            },
            Object {
              "id": 4,
              "image": Object {
                "alt": "Image 2",
                "url": "Example image url",
              },
              "text": "text 4",
              "title": "title 4",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="features"
>
          <div
                    className="features__intro"
          >
                    <h2>
                              Get food fast — not fast food
                    </h2>
                    <p>
                              Hello, we're Omnifood, your new premium food delivery service. We know you're always busy. No time for cooking. So let us take care of that, we're really good at it, we promise!
                    </p>
          </div>
          <div
                    className="features__specs"
          >
                    <GridItem
                              className="specs__span--1--of--4"
                              gridItem={
                                        Object {
                                                  "id": 1,
                                                  "text": "text 1",
                                                  "title": "title 1",
                                                }
                              }
                    />
                    <GridItem
                              className="specs__span--1--of--4"
                              gridItem={
                                        Object {
                                                  "id": 2,
                                                  "text": "text 2",
                                                  "title": "title 2",
                                                }
                              }
                    />
                    <GridItem
                              className="specs__span--1--of--4"
                              gridItem={
                                        Object {
                                                  "id": 3,
                                                  "image": Object {
                                                    "alt": "Image 1",
                                                    "url": "Example image url",
                                                  },
                                                  "text": "text 3",
                                                  "title": "title 3",
                                                }
                              }
                    />
                    <GridItem
                              className="specs__span--1--of--4"
                              gridItem={
                                        Object {
                                                  "id": 4,
                                                  "image": Object {
                                                    "alt": "Image 2",
                                                    "url": "Example image url",
                                                  },
                                                  "text": "text 4",
                                                  "title": "title 4",
                                                }
                              }
                    />
          </div>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="features"
>
          <div
                    className="features__intro"
          >
                    <h2>
                              Get food fast — not fast food
                    </h2>
                    <p>
                              Hello, we're Omnifood, your new premium food delivery service. We know you're always busy. No time for cooking. So let us take care of that, we're really good at it, we promise!
                    </p>
          </div>
          <div
                    className="features__specs"
          >
                    <GridItem
                              className="specs__span--1--of--4"
                              gridItem={
                                        Object {
                                                  "id": 1,
                                                  "text": "text 1",
                                                  "title": "title 1",
                                                }
                              }
                    />
                    <GridItem
                              className="specs__span--1--of--4"
                              gridItem={
                                        Object {
                                                  "id": 2,
                                                  "text": "text 2",
                                                  "title": "title 2",
                                                }
                              }
                    />
                    <GridItem
                              className="specs__span--1--of--4"
                              gridItem={
                                        Object {
                                                  "id": 3,
                                                  "image": Object {
                                                    "alt": "Image 1",
                                                    "url": "Example image url",
                                                  },
                                                  "text": "text 3",
                                                  "title": "title 3",
                                                }
                              }
                    />
                    <GridItem
                              className="specs__span--1--of--4"
                              gridItem={
                                        Object {
                                                  "id": 4,
                                                  "image": Object {
                                                    "alt": "Image 2",
                                                    "url": "Example image url",
                                                  },
                                                  "text": "text 4",
                                                  "title": "title 4",
                                                }
                              }
                    />
          </div>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Features
    featureItems={
        Array [
            Object {
              "id": 1,
              "text": "text 1",
              "title": "title 1",
            },
            Object {
              "id": 2,
              "text": "text 2",
              "title": "title 2",
            },
            Object {
              "id": 3,
              "image": Object {
                "alt": "Image 1",
                "url": "Example image url",
              },
              "text": "text 3",
              "title": "title 3",
            },
            Object {
              "id": 4,
              "image": Object {
                "alt": "Image 2",
                "url": "Example image url",
              },
              "text": "text 4",
              "title": "title 4",
            },
          ]
    }
/>,
}
`;
